web/src/main/java/no/nav/modiapersonoversikt/legacy/sak/service/saf/SafDokumentMapper.kt

```diff
@@ -1,139 +1,132 @@
-        .filter { dato -> dato.datotype == datotype }
-        .map { relevantDato -> relevantDato.dato }
-        .firstOrNull()
-
-private fun getNavn(journalpost: Journalpost) =
-    journalpost.avsenderMottaker?.navn ?: "ukjent"
-
-private fun getHoveddokumentet(journalpost: Journalpost): DokumentInfo =
+        HentBrukersDokumenter.Journalposttype.I -> journalpost.getRelevantDatoForType(HentBrukersDokumenter.Datotype.DATO_REGISTRERT)
+        HentBrukersDokumenter.Journalposttype.U -> listOfNotNull(
+            journalpost.getRelevantDatoForType(HentBrukersDokumenter.Datotype.DATO_EKSPEDERT),
+            journalpost.getRelevantDatoForType(HentBrukersDokumenter.Datotype.DATO_SENDT_PRINT),
+            journalpost.getRelevantDatoForType(HentBrukersDokumenter.Datotype.DATO_JOURNALFOERT)
+        ).firstOrNull()
+        HentBrukersDokumenter.Journalposttype.N -> journalpost.getRelevantDatoForType(HentBrukersDokumenter.Datotype.DATO_JOURNALFOERT)
+        else -> LocalDateTime.now()
+    } ?: LocalDateTime.now()
+
+private fun HentBrukersDokumenter.Journalpost.getRelevantDatoForType(type: HentBrukersDokumenter.Datotype): LocalDateTime? {
+    return this.relevanteDatoer
+        .orEmpty()
+        .filterNotNull()
+        .find { it.datotype == type }
+        ?.dato
+        ?.value
+}
+
+private fun getVedlegg(journalpost: HentBrukersDokumenter.Journalpost): List<Dokument> =
+    getElektroniskeVedlegg(journalpost).plus(getLogiskeVedlegg(journalpost))
+
+private fun getElektroniskeVedlegg(journalpost: HentBrukersDokumenter.Journalpost): List<Dokument> =
+    journalpost.dokumenter
+        .orEmpty()
+        .subList(SafServiceImpl.VEDLEGG_START_INDEX, journalpost.dokumenter?.size ?: 0)
+        .filterNotNull()
+        .map { dok -> Dokument().fraSafDokumentInfo(dok) }
+
+private fun getHoveddokumentet(journalpost: HentBrukersDokumenter.Journalpost): HentBrukersDokumenter.DokumentInfo =
     journalpost.dokumenter?.get(0) ?: throw RuntimeException("Fant sak uten hoveddokument!")
 
-private fun Dokument.fraSafDokumentInfo(dokumentInfo: DokumentInfo): Dokument {
+private fun getLogiskeVedlegg(journalpost: HentBrukersDokumenter.Journalpost): List<Dokument> =
+    getHoveddokumentet(journalpost).logiskeVedlegg
+        .filterNotNull()
+        .map { logiskVedlegg -> Dokument().fraSafLogiskVedlegg(logiskVedlegg) }
+
+private fun Dokument.fraSafDokumentInfo(dokumentInfo: HentBrukersDokumenter.DokumentInfo): Dokument {
     tittel = dokumentInfo.tittel
     dokumentreferanse = dokumentInfo.dokumentInfoId
     isKanVises = true
     isLogiskDokument = false
     variantformat = getVariantformat(dokumentInfo)
-    skjerming = getSkjerming(dokumentInfo)
-    dokumentStatus =
-        dokumentInfo.dokumentStatus?.let { Dokument.DokumentStatus.valueOf(it) } ?: Dokument.DokumentStatus.FERDIGSTILT
-
+    skjerming = getSkjerming(dokumentInfo)?.toString()
+    dokumentStatus = getDokumentStatus(dokumentInfo)
```

Comment:
Kult, visste ikke at man bare kunne skrive det sånn uten å si `this` i en extension function 👌🏼 Ser at Kotlin parser det som om vi bruker `set`-funksjonen på verdiene.